docs/stdout-exporter-design.md

```diff
@@ -0,0 +1,223 @@
+
+The specification states: Any retry logic that is required by the exporter is the responsibility of the exporter. The default SDK SHOULD NOT implement retry logic, as the required logic is likely to depend heavily on the specific protocol and backend the spans are being sent to.
+
+The MetricsExporter is called through the Controller in the SDK data path. The exporter will either be called on a regular interval in the case of a push controller or through manual calls in the case of a pull controller.
+
+### `Shutdown()`
+
+Shuts down the exporter. Called when SDK is shut down. This is an opportunity for exporter to do any cleanup required.
+
+`Shutdown` should be called only once for each `Exporter` instance. After the call to `Shutdown` subsequent calls to `Export` are not allowed and should return a `Failure` result.
+
+`Shutdown` should not block indefinitely (e.g. if it attempts to flush the data and the destination is unavailable). Language library authors can decide if they want to make the shutdown timeout configurable.
+
+In the StdoutExporter there is no cleanup to be done, so there is no need to use the timeout within the `Shutdown` function as it will never be blocking.
+
+```
+class StdoutMetricsExporter final : public sdkmeter::MetricsExporter
+{
+
+private: 
+    bool isShutdown = false;
+    
+public:
+    sdkmeter::ExportResult Export(
+        const Collection<Record> batch) noexcept
+    {
+    
+        Time startTime = time.now();
+        
+        if(isShutdown)
+        {
+            return sdktrace::ExportResult::kFailure;
+        }
+       
+        for (auto &metric : batch)
+        {
+                            
+            if (metric != nullptr)
+            {
+        
+                if(metric.AggregationType == CounterAggregator) {
+                    std::cout << "{"
+                              << "\n  name          : " << metric->GetName()
+                              << "\n  labels        : " << metric->GetLabels()
+                              << "\n  sum           : " << metric->Value[0] << "\n}"
+                }
+                else if(metric.AggregationType == SketchAggregator) {
+                    // Similarly print data
+                }
+                // Other Aggreagators will also be handeled,
+                // Counter, Gauge, MinMaxSumCount, Sketch, Histogram,
+                // and Exact Aggreagtors        
+            }
+            
+            if(time.now() - startTime > 30s) {
+                return sdktrace::ExportResult::kFailure;
+            }
+        }
+    
+        return sdkmeter::ExportResult::kSuccess;
+    }
+
+    void Shutdown(microseconds timeout = microseconds(0)) noexcept
+    {
+        isShutdown = true;
```

Comment:
This seems to be against what has been stated above "After the call to `Shutdown` subsequent calls to `Export` are not allowed and should return a `Failure` result.".

Reply:
At the top of the export function I return failure if the exporter is shutdown. Is there a different way to implement this wording from the spec?

Reply:
I think it is fine to leave it as is. This might introduce some race condition but I don't think Console Exporter has to cover such case.